### PR TITLE
Improved cheat sheet

### PR DESCRIPTION
**Pull request by walseb**
I thought doing #128 would be a good exercise so I tried adding a few patterns and operators I have picked up on. I also added in the small table in the beginning of "Rhine: FRP with Type-Level Clocks" to help with the different terms if that's okay. Please correct me if anything here is wrong!

### CODE REVIEW

File: CHEATSHEET.md [hunk 1]
```diff
@@ -19,13 +24,56 @@ If you miss something, feel free to open an issue or pull request.
...
 
-### Common combinators in action
+### Common combinators
+
+#### SN composition
+
+##### Sequential composition
+| Operator        | Description                                                                                | Simplified Type Signature                                                |
+|-----------------|--------------------------------------------------------------------------------------------|--------------------------------------------------------------------------|
+| `>>>` and `<<<` | Arrow composition                                                                          | `Category cat => cat a b -> cat b c -> cat a c`                          |
```
Comment (turion) [inline CHEATSHEET.md]:
```suggestion
| `>>>` and `<<<` | Arrow composition                                                                          | `ClSF m cl a b -> ClSF m cl b c -> ClSF m cl a c`                        |
```

File: CHEATSHEET.md [hunk 2]
```diff
@@ -19,13 +24,56 @@ If you miss something, feel free to open an issue or pull request.
...
+| `>>>^`          | Postcompose a signal network with a pure function                                          | `SN m cl a b -> (b -> c) -> SN m cl a c`                                 |
+
+##### Parallel composition
+| Operator   | Clock Type | Input Type | Output Type | Resulting Type                             |
+|------------|------------|------------|-------------|--------------------------------------------|
+| `****`     | Same       | Different  | Different   | `SN m cl (a, c) (b, d)`                    |
+| `++++`     | Different  | Same       | Different   | `SN m (ParClock m clL clR) a (Either b c)` |
+| `\|\|\|\|` | Different  | Same       | Same        | `SN m (ParClock m clL clR) a b`            |
+
+##### Parallel rhine composition
```
Comment (turion) [inline CHEATSHEET.md]: Is it possible to explain `||@` and `@||` together, i.e. describe how `rhine1 ||@ schedule @|| rhine2` work? If you ever need to talk about the intermediate type `RhineParallelAndSchedule`, make clear that it is only necessary to define the "ternary" operator `_ ||@ _ @|| _`.

Comment (turion) [reply CHEATSHEET.md]: Keep in mind that this is a cheatsheet, which one could ideally print on one or two pages, or open in a separate tab without scrolling around too much. Extra documentation has to go elsewhere. But feel free to start a `doc` directory and collect longer texts like this there!

File: CHEATSHEET.md [hunk 3]
```diff
@@ -19,13 +24,56 @@ If you miss something, feel free to open an issue or pull request.
...
+To compose two `Rhine`s in parallel we need a shared `Schedule` between those two. To share a `Schedule` between two `Rhine`s we need to first construct a `RhineParallelAndSchedule` out of one `Rhine` and one `Schedule`. To do this we can use the constructor `RhineParallelAndSchedule` or the aliases `||@` or `++@`. Then we need to combine that resulting `RhineParallelAndSchedule` with another `Rhine` using `@||` or `@++`
+
+| Operator   | Resulting Type                                     |
+|------------|----------------------------------------------------|
+| `@\|\|`    | `Rhine m (ParallelClock m clL clR) a b`            |
+| `@++`      | `Rhine m (ParallelClock m clL clR) a (Either b c)` |
+
+###### Relation between |||| and @||, ++++ and @++
```
Comment (turion) [inline CHEATSHEET.md]: Same as the comment above. Longer text and explanation has to go to another place than the cheatsheet. Maybe there could be a file `docs/Operators.md` that explains at length what kinds of operators there are, how they work, and when to use them? As a first approximation, you can simply move these texts there.

**Comment by walseb** [comment]
Cool! Yeah I also thought that maybe the longer texts shouldn't be in the cheat sheet. I will try to work on this next weekend and maybe by then I will have more stuff to add

**Comment by turion** [comment]
I took the liberty and updated the cheatsheet and added a new file. Thanks again!

**Comment by walseb** [comment]
Oh sorry for never following up on this. Here are some old notes I found of the improvements to the cheat sheet I was working on in case you are interested. Feel free to do whatever you want with them. But I'm not sure how readable some of the diagrams are or how accurate they are
``` markdown
###### RhineParallelAndSchedule constructors
These operators are aliases of the constructor `RhineParallelAndSchedule`
| Operator |
|----------|
| `||@`    |
| `++@`    |
```

``` markdown
x-axis is time, the end of lines means the function returned a result
# Data parallel      
              (got result)              
    a -> b         |               
    (process 1)    v              
   ----------------\ -------------------------
                   |
                   | Result               
                   |  b                    
                   \-------------------------
                        /---------------- /--
                        | Result          | 
                        |  d              | 
    a -> b              |                 |
    (process 2)         |                 |
   ---------------------/ ----------------/ --
                           
# Time parallel                                        
    a -> b                        
    (process 1)                  
   ----------------\ -------------------------
                   |
                   | Result 
                   |  b 
                   \--- /---------------- /--
                        |                 |
    a -> b              |                 |
    (process 2)         |                 |
   ---------------------/ ----------------/ --

# Synchronous
     a -> b                        a -> b
    ---------\                ---------\   
             |                         |
      Result |                  Result |
       b     |                   b     | 
             | b -> c                  | c -> d
             \----------------         \--------------
                                
# Sequential              
     a -> b
    -------------\ a -> b
                 ---------------\ a -> b
                 |              ---------------\ a -> b
      Resampling |                             ----------\  a -> b
       b -> c    |                             |         -------------
                 | c -> d                      |         |
                 \---------------------------------      |
                                               |         |
                                    Resampling |         |
                                     b -> c    |         |
                                               | c -> d  |
                                               \---------|---------
                                                         |
                                                         |
                                              Resampling |
                                               b -> c    |
                                                         | c -> d
                                                         \------------
```

**Pull request merged**